### PR TITLE
Bump dependencies and php version

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Need https://github.com/sonata-project/dev-kit/pull/2038

File: src/FlashMessage/FlashManager.php [hunk 1]
```diff
@@ -22,27 +22,12 @@
...
-     * @var array<string, string>
-     */
-    private array $cssClasses;
-
     /**
      * @param array<string, array<string>> $types      Sonata flash message types array (defined in configuration)
      * @param array<string, string>        $cssClasses Css classes associated with $types
      */
-    public function __construct(RequestStack $requestStack, array $types, array $cssClasses)
+    public function __construct(private RequestStack $requestStack, private array $types, private array $cssClasses)
```
Comment (jordisala1991) [inline src/FlashMessage/FlashManager.php]: For construct with more than 1 argument , I have been doing multiline, wdyt? for readability

File: src/Node/TemplateBoxNode.php [hunk 2]
```diff
@@ -19,18 +19,14 @@
...
-    protected bool $enabled;
-
     /**
      * @param AbstractExpression $message Node message to display
      * @param bool               $enabled Is Symfony debug enabled?
      * @param int|null           $lineno  Symfony template line number
      * @param string|null        $tag     Symfony tag name
      */
-    public function __construct(AbstractExpression $message, bool $enabled, ?int $lineno = null, ?string $tag = null)
+    public function __construct(AbstractExpression $message, protected bool $enabled, ?int $lineno = null, ?string $tag = null)
```
Comment (jordisala1991) [inline src/Node/TemplateBoxNode.php]: this multiline would be more readable

File: src/FlashMessage/FlashManager.php (hunk 1 above)
Comment (VincentLanglet) [reply src/FlashMessage/FlashManager.php]: Sure done

File: src/Node/TemplateBoxNode.php (hunk 2 above)
Comment (VincentLanglet) [reply src/Node/TemplateBoxNode.php]: Done

**Comment by jordisala1991** [comment]
Also you can modify services definition. An example is what I did for form-extensions or block bundle

File: composer.json [hunk 3]
```diff
@@ -40,13 +40,13 @@
...
         "rector/rector": "^0.12",
-        "symfony/browser-kit": "^4.4 || ^5.3 || ^6.0",
-        "symfony/dependency-injection": "^4.4 || ^5.3 || ^6.0",
-        "symfony/framework-bundle": "^4.4 || ^5.3 || ^6.0",
-        "symfony/http-foundation": "^4.4 || ^5.3 || ^6.0",
+        "symfony/browser-kit": "^5.4 || ^6.0",
+        "symfony/dependency-injection": "^5.4 || ^6.0",
+        "symfony/framework-bundle": "^5.4 || ^6.0",
+        "symfony/http-foundation": "^5.4 || ^6.0",
         "symfony/http-kernel": "^4.4.13 || ^5.3 || ^6.0",
```
Comment (jordisala1991) [inline composer.json]: you didnt change this one

File: composer.json [hunk 4]
```diff
@@ -40,13 +40,13 @@
...
         "symfony/phpunit-bridge": "^6.0",
-        "symfony/twig-bundle": "^4.4 || ^5.3 || ^6.0",
+        "symfony/twig-bundle": "^5.4 || ^6.0",
         "vimeo/psalm": "^4.7.2"
     },
     "conflict": {
```
Comment (jordisala1991) [inline composer.json]: Should we remove this conflict? Sonata 3 wont be compatible with this anyway

File: composer.json (hunk 3 above)
Comment (VincentLanglet) [reply composer.json]: Fixed

File: composer.json (hunk 4 above)
Comment (VincentLanglet) [reply composer.json]: Done

File: src/Bridge/Symfony/Resources/config/flash.php [hunk 5]
```diff
@@ -24,23 +24,21 @@
...
-    // Use "param" function for creating references to parameters when dropping support for Symfony 5.1
     $containerConfigurator->services()
 
         ->set('sonata.twig.flashmessage.manager', '%sonata.twig.flashmessage.manager.class%')
             ->public()
             ->tag('sonata.status.renderer')
             ->args([
-                new ReferenceConfigurator('request_stack'),
+                service('request_stack'),
                 [],
```
Comment (jordisala1991) [inline src/Bridge/Symfony/Resources/config/flash.php]: if this is an abstract argument it can be replaced too, abstract_arg('explanation of argument')

Comment (VincentLanglet) [reply src/Bridge/Symfony/Resources/config/flash.php]: Done

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**